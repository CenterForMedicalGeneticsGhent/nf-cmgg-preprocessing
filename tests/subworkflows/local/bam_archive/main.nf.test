nextflow_workflow {

    name "Test Workflow BAM_ARCHIVE"
    script "subworkflows/local/bam_archive/main.nf"
    workflow "BAM_ARCHIVE"

    test("Should run without failures") {

        when {
            params {
                // define parameters here. Example:
                // outdir = "tests/results"
            }
            workflow {
                """
                input[0] = Channel.of(
                    [
                        [ id:'test_cram', single_end:false ], // meta map
                        file(params.test_data['homo_sapiens']['illumina']['test_cram'], checkIfExists: true),
                        file(params.test_data['homo_sapiens']['illumina']['test_cram_index'], checkIfExists: true)
                    ],
                    [
                        [ id:'test_bam', single_end:false ], // meta map
                        file(params.test_data['homo_sapiens']['illumina']['test_bam'], checkIfExists: true),
                        file(params.test_data['homo_sapiens']['illumina']['test_bam_index'], checkIfExists: true)
                    ]
                )

                input[1] = Channel.value(
                    [
                        [ id:'test_fasta' ], // meta map
                        file(params.test_data['homo_sapiens']['genome']['genome_fasta'], checkIfExists: true),
                        file(params.test_data['homo_sapiens']['genome']['genome_fasta_fai'], checkIfExists: true)
                    ]
                )
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()
        }

    }

}
