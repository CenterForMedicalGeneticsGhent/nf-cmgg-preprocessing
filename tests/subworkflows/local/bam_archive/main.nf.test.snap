{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test_bam",
                            "single_end": false
                        },
                        "test_bam.cram:md5,2fdf77094cae185f31c92700f107ea17",
                        "test_bam.cram.crai:md5,5c3b266b82be397c565b8e7ea14ccd7c"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test_cram",
                            "single_end": false
                        },
                        "test.cram.md5:md5,110faa3bcad1ae0381f0c6fdfd78c4d4"
                    ],
                    [
                        {
                            "id": "test_bam",
                            "single_end": false
                        },
                        "test_bam.cram.md5:md5,04afb90cd2568d2469d641cced7d4b0d"
                    ]
                ],
                "2": [
                    "versions.yml:md5,1f16bdc26750fb00d72ba015a19d2b00",
                    "versions.yml:md5,1cc0a1704d11534645c0c1c1d7dbcdb3",
                    "versions.yml:md5,1f16bdc26750fb00d72ba015a19d2b00"
                ],
                "checksum": [
                    [
                        {
                            "id": "test_cram",
                            "single_end": false
                        },
                        "test.cram.md5:md5,110faa3bcad1ae0381f0c6fdfd78c4d4"
                    ],
                    [
                        {
                            "id": "test_bam",
                            "single_end": false
                        },
                        "test_bam.cram.md5:md5,04afb90cd2568d2469d641cced7d4b0d"
                    ]
                ],
                "cram_crai": [
                    [
                        {
                            "id": "test_cram",
                            "single_end": false
                        },
                        "/nf-cmgg/nf-cmgg-test-datasets/main/data/genomics/homo_sapiens/illumina/cram/test.cram",
                        "/nf-cmgg/nf-cmgg-test-datasets/main/data/genomics/homo_sapiens/illumina/cram/test.cram.crai"
                    ],
                    [
                        {
                            "id": "test_bam",
                            "single_end": false
                        },
                        "test_bam.cram:md5,2fdf77094cae185f31c92700f107ea17",
                        "test_bam.cram.crai:md5,5c3b266b82be397c565b8e7ea14ccd7c"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,1f16bdc26750fb00d72ba015a19d2b00",
                    "versions.yml:md5,1cc0a1704d11534645c0c1c1d7dbcdb3",
                    "versions.yml:md5,1f16bdc26750fb00d72ba015a19d2b00"
                ]
            }
        ],
        "timestamp": "2023-06-14T08:50:10+0000"
    }
}
